testdata/src/go.uber.org/arrays/arrays.go

```diff
@@ -155,3 +155,32 @@ func test2dArrayAssignment() *int {
 	twodArr = nilableTwodArr // TODO: an error should be reported here since we are assigning a (default) deeply nilable array 'nilableTwodArr' into a declared deeply nonnil array 'twodArr'
 	return twodArr[0][0]     //want "returned"
 }
+
+// Test a case where we declare a type alias for an array and then range over it.
+
+type blocks [42]int
+
+type blockSlice []int
+
+// nonnil(aPtr) nonnil(a) nonnil(bPtr) nonnil(b)
```

Comment:
Should work fine even with a single `nonnil()`

`// nonnil(aPtr, a, bPtr, b)`

Reply:
I was experimenting so I separated the annotations for quick edits. 

But yes, should've cleaned that up. Thanks!